src/main.cpp

```diff
@@ -31,21 +59,33 @@ int main(int argc, char* argv[])
         printf("Renderer is null\n");
     }
 
-    backup_camera->init_screen_settings(window, 0, atoi(argv[5]), atoi(argv[6]));
-    backup_camera->init_graphics(renderer);
+    backupCamera_->init_screen_settings(window_, 0, atoi(argv[5]), atoi(argv[6]));
+    backupCamera_->init_graphics(renderer);
     printf("Starting threads\n");
-    backup_camera->start_threads();
+    backupCamera_->start_threads();
+    vector<char> eventsToHandle;
+    //bool quitSignal_  = false;
 
-    while (backup_camera->process_events())
+    while (!quitSignal_)
     {
-        if (backup_camera->update())
+        eventsToHandle = backupCamera_->process_events();
+
+        if (backupCamera_->update())
         {
             SDL_RenderPresent(renderer);
             SDL_SetRenderDrawColor(renderer, 0, 0, 0, 0);
             SDL_RenderClear(renderer);
         }
+
+        while (!eventsToHandle.empty())
+        {
+            handleEvent(eventsToHandle.back());
+            eventsToHandle.pop_back();
```

Comment:
Its for handling the fullscreen and quit events at the moment based on a list of events returned by backup camera. I think im going to try and get rid of it since im probably just overcomplicating this